### PR TITLE
Add StatusEvent.sequence column for properly ordering status events

### PR DESCRIPTION
**Pull request by richard-dds**
Since status events were being ordered by `time_created`, events created in the same transaction had nondeterministic ordering. Using a sequence ensures that the events are always ordered properly.

### CODE REVIEW

File: requests_queue/models/status_event.py [hunk 1]
```diff
@@ -12,4 +13,5 @@ class StatusEvent(Base):
     id = Id()
     new_status = Column(String())
     time_created = Column(DateTime(timezone=True), server_default=func.now())
-    request_id = Column(UUID(as_uuid=True), ForeignKey('requests.id'))
+    request_id = Column(UUID(as_uuid=True), ForeignKey('requests.id', ondelete='CASCADE'))
```
Comment (patricksmithdds) [inline requests_queue/models/status_event.py:16]: Do we want to add the `ondelete='CASCADE'` here?

Comment (richard-dds) [reply requests_queue/models/status_event.py:16]: I think so. No reason in keeping status events around if the request has been deleted, right?

Comment (patricksmithdds) [reply requests_queue/models/status_event.py:16]: Do we have any established data retention policies? I assumed requests and their statuses would not actually be deleted from the database for accountability purposes, but maybe that's overcomplicating things.

Comment (richard-dds) [reply requests_queue/models/status_event.py:16]: Yeah, I agree that it's unlikely that a request would ever be deleted. But if it was deleted, its status history would probably become useless.

**Review by patricksmithdds — APPROVED** [review]

**Pull request merged**